tests/macsec/test_lc_reboot.py

```diff
@@ -0,0 +1,36 @@
+'''This script is to test the line card reboot behavior for SONiC.
+
+Step 1: Configure Macsec between neighbor and DUT
+Step 2: Save config on DUT and reboot the line card
+Step 3: Verify macsec connection is re-established after reboot
+
+'''
+
+import logging
+import pytest
+from tests.common import reboot
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t2')
+]
+
+
+@pytest.mark.reboot
+def test_chassis_reboot(duthost, localhost):
```

Comment:
since this is T2 specific test, suggest to use `duthosts` and `enum_rand_one_dut_per_hwsku` so test can run on all different sku. Current design will choose only 1 random dut from LC/sup

Reply:
and can you change the function name to test_lc_reboot

Reply:
@wenyiz2021 I don't see a fixture named enum_rand_one_dut_per_hwsku in the codebase.  I am also not understanding your ask.  This test only runs on one DUT, are you saying using duthost has a selection process for choosing the DUT that is not appropriate for this test case?

I will update the test name.

Reply:
You can use `enum_rand_one_per_hwsku_hostname`.

Reply:
I mean if you use duthost, it'll choose only 1 random dut, it is ok but not comprehensive, if you use `enum_rand_one_per_hwsku_hostname`  it'll run for every LC that has different sku, also run on sup, so it'll test both chassis reboot and LC reboot, please try this

Reply:
@wenyiz2021 So I'm clear, you're proposing that I should iterate through enum_rand_one_per_hwsku_hostname and do both lc and chassis reboot tests as one test?

Reply:
I said it wrongly.
still separate sup and LC reboot.
use `enum_supervisor_dut_hostname` for sup reboot test.
`enum_rand_one_per_hwsku_frontend_hostname` will only choose the LCs that has different sku

Reply:
I'm proposing to have them in one file

Reply:
`enum_rand_one_per_hwsku_frontend_hostname` is not supposed to get you a SUP DUT. `enum_rand_one_per_hwsku_hostname` may get you a SUP. Can you confirm whether you are using the latter one?

Reply:
@BYGX-wcr I am using duthost to get the lc and duthosts and enum_supervisor_dut_hostname to get the sup currently.

Reply:
The duthost may get you a SUP DUT. I thought that you have changed to use `enum_rand_one_per_hwsku_frontend_hostname` for lc reboot. Can you please try using the `enum_rand_one_per_hwsku_frontend_hostname` and post the error message if you encounter one?

Reply:
One possible reason why your `enum_rand_one_per_hwsku_frontend_hostname` will return a SUP DUT is that in your inventory file, your SUP DUT does not carry the attribute "card_type: supervisor".

Reply:
@BYGX-wcr I ran the test using enum_rand_one_per_hwsku_hostname, but I get this error:

__________________________________ ERROR at setup of test_chassis_reboot[256_XPN_SCI-vlab-t2-sup] ___________________________________

request = <SubRequest 'enum_rand_one_per_hwsku_hostname' for <Function test_chassis_reboot[256_XPN_SCI-vlab-t2-sup]>>

    @pytest.fixture(scope="module")
    def enum_rand_one_per_hwsku_hostname(request):
>       return request.param
E       AttributeError: 'SubRequest' object has no attribute 'param'

request    = <SubRequest 'enum_rand_one_per_hwsku_hostname' for <Function test_chassis_reboot[256_XPN_SCI-vlab-t2-sup]>>

conftest.py:1639: AttributeError


That enum pulled a sup, but I have not had duthost return a sup when using that fixture.

Reply:
I also checked my inventory file and the sup is marked card_type:supervisor.

Reply:
Why are you using `enum_rand_one_per_hwsku_hostname`? As I suggested, for test_lc_reboot, get duthost by using `duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]`. For test_chassis_reboot, what you are doing with the fixture is OK. Yet, please address Judy's comment regarding the test logic below.

Reply:
I'll try that.  I am working on the other comments, I have not said anything about them as they are in progress and there is nothing to say.

Reply:
I am using enum_rand_one_per_hwsku_frontend_hostname but sometimes it gets a non-macsec DUT.  Is there a way to have it select a new DUT or force it to pick a macsec device?

Reply:
Supposedly, all frontend DUTs in your testbed should be configured to enable MACsec. How can you get a non-macsec frontend DUT? Maybe check your testbed configuration?

For `enum_rand_one_per_hwsku_frontend_hostname`, skipping the DUT may skip all frontend DUTs if all the DUTs belong to the same SKU.

Reply:
One way to make the test more robust is to use `enum_frontend_dut_hostname`, this fixture will feed all frontend DUTs into the test one-by-one. Then your test can check whether a DUT is MACsec-enabled. If not, you can skip it. However, by using such a fixture, it will increase the test running time. So, please check your testbed configuration and triage the problem of `enum_rand_one_per_hwsku_frontend_hostname` first.

Reply:
My understanding is macsec is only enabled for the T2 to T3 links, not the T2 to T1 links.  DUT02 is the T2 to T1 connections.  So it does not have macsec enabled.  This is how we test on the manual physical testbed as well.

Reply:
OK. Then try using `enum_frontend_dut_hostname`.

Reply:
I have adjusted to use this fixture.